File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() < 1)
            return 0;
        
        set<int> numSet(nums.begin(), nums.end());
        vector<int> vec(numSet.begin(), numSet.end());

        int ans = 1, len = 1, prev = 0, curr = 1;

        while(curr < vec.size()) {
            if(vec[curr] - vec[prev] == 1)
                len++;
            else {
                ans = max(ans, len);
                len = 1;
            }
            prev++, curr++;
        }

        ans = max(ans, len);
        return ans;
    }
};

File: Data Structures & Algorithms/sliding-window-maximum/submission-0.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int n = nums.length;
        int[] res = new int[n-k+1];

        for(int i=0; i<=n-k; i++) {
            int max = nums[i];
            for(int j=i; j<i+k; j++) {
                max = Math.max(max, nums[j]);
            }
            res[i] = max;
        }

        return res;
    }
}
